state-utils.md

```diff
@@ -343,6 +394,43 @@ The `"rlp"` key loads the block information.
            <txType> AccessList </txType>
            ...
          </message>
+
+    rule [[ #getTxData( TXID ) => DynamicFeeTxData(TN, TPF, TM, TG, TT, TV, DATA, CID, TA) ]]
+         <message>
+           <msgID>         TXID </msgID>
+           <txNonce>       TN   </txNonce>
+           <txGasLimit>    TG   </txGasLimit>
+           <to>            TT   </to>
+           <value>         TV   </value>
+           <data>          DATA </data>
+           <txChainID>     CID  </txChainID>
+           <txAccess>      TA   </txAccess>
+           <txPriorityFee> TPF  </txPriorityFee>
+           <txMaxFee>      TM   </txMaxFee>
+           <txType> DynamicFee </txType>
+           ...
+         </message>
+
+    syntax Int ::= #effectiveGasPrice( Int ) [function]
+ // ---------------------------------------------------
+    rule [[ #effectiveGasPrice( TXID )
+         => #if notBool Ghasbasefee << SCHED >>
+                orBool TXTYPE ==K Legacy
+                orBool TXTYPE ==K AccessList
```

Comment:
I would recommend using parens around `notBool Ghasbasefee << SCHED >>`

Also just want to mention that `#if_#then_#else_#fi` doesn't behave very well for symbolic execution, so often it can be better to split the rule into two instead if there is any chance that the condition in the `#if_...` is going to be symbolic. There is no risk of that happening here though, because this is `driver.md` which isn't really used for symbolic execution.

Reply:
Noted. Although I do see quite a lot of `#if_#then...` being used in evm.md. The main concern is that each branch of the statement has to be well defined, is that right?

Reply:
This is just a suggestion.
I would have a rule `#computeGasPrice` like 
```
    syntax EthereumCommand ::= "#computeGasPrice"
 // ---------------------------------------------
    rule <k> #computeGasPrice => . ... </k> <schedule> SCHED </schedule> requires notBool Ghasbasefee << SCHED >>

    rule <k> #computeGasPrice => . ... </k>
         <schedule> SCHED </schedule>
         <baseFeePerGas> BF </baseFeePerGas>
         <txPending> ListItem(TXID:Int) ... </txPending>
         <message>
           <msgID>         TXID </msgID>
           <txMaxFee>      MF   </txMaxFee>
           <txPriorityFee> PF   </txPriorityFee>
           <txGasPrice> _ => BF +Int minInt(PF, MF -Int BF) </txGasPrice>
           ...
         </message>
      requires Ghasbasefee << SCHED >>
```

and have it called before `loadTx` 
```
    rule <k> startTx => #computeGasPrice ~> loadTx(...) ... </k>
```

`loadTx` would then remain unchaged, and update the `<gasPrice>` cell with the value from `<txGasPrice>`, which is the `effectiveGasPrice`